### PR TITLE
aws ipi ccm disconnected private

### PR DESCRIPTION
**Pull request by huali9**
add the workflow "cucushift-installer-rehearse-aws-ipi-ccm-disconnected-private" and provision & deprovision chains

@jhou1 @sunzhaohua2 @miyadav Please help to take a look, thanks!

### CODE REVIEW

**Comment by huali9** [comment]
> @huali9 Did the make update ran successfully and generated any metadata(periodics yaml) ?

Yes, make update ran successfully, but didn't generate any metadata file for me.

**Comment by huali9** [comment]
@jianlinliu I am trying to add a workflow combining ccm and disconnected-private, I copied https://github.com/openshift/release/tree/master/ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected to https://github.com/openshift/release/tree/master/ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/ccm and made some changes (add ref: ipi-conf-techpreview).
It ran three times, they all failed but failed at different steps. Can you please take a look? Thanks!
1st: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29451/rehearse-29451-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.10-installer-rehearse-aws/1536664016405925888 `installer-rehearse-aws-ipi-conf-mirror` failed.
2nd: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29451/rehearse-29451-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.10-installer-rehearse-aws/1536914290261364736 `installer-rehearse-aws-ipi-install-install-aws` failed
3rd: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29451/rehearse-29451-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.10-installer-rehearse-aws/1536940722165387264  `installer-rehearse-aws-aws-provision-vpc-disconnected` failed

**Comment by jianlinliu** [comment]
For the first 2 job, they are from network flake, similar issues are discussed in https://coreos.slack.com/archives/CF8SMALS1/p1655268548663719?thread_ts=1655261850.199909&cid=CF8SMALS1

For the 3rd one, seem like aws vpc quota limit is reached, https://coreos.slack.com/archives/CH76YSYSC/p1655272626153529

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/ccm/disconnected/private/cucushift-installer-rehearse-aws-ipi-ccm-disconnected-private-workflow.metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"path": "cucushift/installer/rehearse/aws/ipi/ccm/disconnected/private/cucushift-installer-rehearse-aws-ipi-ccm-disconnected-private-workflow.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/ccm/disconnected/private/cucushift-installer-rehearse-aws-ipi-ccm-disconnected-private-workflow.metadata.json:2]: I suggest to place it in `aws/ipi/disconnected/private/ccm`

**Comment by huali9** [comment]
> For the first 2 jobs, they are from network flake, similar issues are discussed in https://coreos.slack.com/archives/CF8SMALS1/p1655268548663719?thread_ts=1655261850.199909&cid=CF8SMALS1
> 
> For the 3rd one, seem like aws vpc quota limit is reached, https://coreos.slack.com/archives/CH76YSYSC/p1655272626153529

Good to know, thanks!

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/ccm/disconnected/private/cucushift-installer-rehearse-aws-ipi-ccm-disconnected-private-workflow.metadata.json (hunk 1 above)
Comment (huali9) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/ccm/disconnected/private/cucushift-installer-rehearse-aws-ipi-ccm-disconnected-private-workflow.metadata.json:2]: ok, thanks! Let me change it.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/ccm/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-ccm-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    - chain: aws-provision-bastionhost
+    - ref: aws-provision-cco-manual-users-static
+    - chain: mirror-images-by-oc-adm
+    - ref: ipi-conf
+    - ref: ipi-conf-mirror
+    - ref: ipi-conf-aws-custom-vpc
+    - ref: ipi-conf-manual-creds
+    - ref: ipi-conf-aws
+    - ref: ipi-install-monitoringpvc
+    - ref: ipi-conf-techpreview
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/ccm/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-ccm-provision-chain.yaml:14]: why not using `ccm-conf-apply-feature-gate` like other install workflow?

**Comment by huali9** [comment]
Yeah, success last night!
@jhou1 @sunzhaohua2 @miyadav @jianlinliu Please help to take a look, thanks!
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29451/rehearse-29451-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.10-installer-rehearse-aws/1537071320120954880

**Comment by sunzhaohua2** [comment]
> why not using ccm-conf-apply-feature-gate like other install workflow?

@jianlinliu  ccm-conf-apply-feature-gate is for post installation enable ccm, ipi-conf-techpreview is for during installation enable ccm. We want to cover the both scenarios.

**Pull request merged**